tools/ci_build/github/azure-pipelines/linux-rocm-ci-pipeline.yml

```diff
@@ -0,0 +1,238 @@
+
+  - task: CmdLine@2
+    inputs:
+      script: |
+        cd $(Build.BinariesDirectory)/Release
+        find -executable -type f > $(Build.BinariesDirectory)/Release/perms.txt
+    displayName: 'Find Executable Files'
+
+  - task: PublishPipelineArtifact@0
+    displayName: 'Publish Pipeline Artifact'
+    inputs:
+      artifactName: 'drop-linux'
+      targetPath: '$(Build.BinariesDirectory)/Release'
+
+  - template: templates/explicitly-defined-final-tasks.yml
+
+- job: Linux_Test
+  workspace:
+    clean: all
+  pool: AMD-GPU
+  dependsOn:
+  - Linux_Build
+  timeoutInMinutes: 120
+
+  steps:
+  - task: DownloadPipelineArtifact@2
+    displayName: 'Download Pipeline Artifact'
+    inputs:
+      buildType: 'current'
+      artifactName: 'drop-linux'
+      targetPath: '$(Build.BinariesDirectory)/Release'
+
+  - checkout: self
+    clean: true
+    submodules: recursive
+
+  - template: templates/get-docker-image-steps.yml
+    parameters:
+      Dockerfile: tools/ci_build/github/linux/docker/rocm-ci-pipeline-env.Dockerfile
+      Context: tools/ci_build/github/linux/docker
+      DockerBuildArgs: "--build-arg ROCM_VERSION=$(RocmVersion)$(RocmVersionPatchSuffix)"
+      Repository: onnxruntimerocm-cibuild-rocm$(RocmVersion)
+
+  - task: CmdLine@2
+    inputs:
+      script: |
+        docker run --rm \
+          --security-opt seccomp=unconfined \
+          --shm-size=1024m \
+          --device=/dev/kfd \
+          --device=/dev/dri/renderD$DRIVER_RENDER \
+          --group-add $(video) \
+          --group-add $(render) \
+          --user $UID:$(id -g $USER) \
+          --volume $(Build.SourcesDirectory):/onnxruntime_src \
+          --volume $(Build.BinariesDirectory):/build \
+          --volume /data/models:/build/models:ro \
+          --workdir /build/Release \
+          onnxruntimerocm-cibuild-rocm$(RocmVersion) \
+          /bin/bash -c "
+            set -ex; \
+            xargs -a /build/Release/perms.txt chmod a+x; \
+            python /onnxruntime_src/tools/ci_build/build.py \
+              --config Release \
+              --cmake_extra_defines \
+                CMAKE_HIP_COMPILER=/opt/rocm/llvm/bin/clang++ \
+                onnxruntime_BUILD_KERNEL_EXPLORER=ON \
+                CMAKE_HIP_ARCHITECTURES=gfx90a \
+              --mpi_home /opt/ompi \
+              --use_rocm \
+              --rocm_version=$(RocmVersion) \
+              --rocm_home /opt/rocm \
+              --nccl_home /opt/rocm \
+              --enable_nccl \
+              --build_dir /build \
+              --build_shared_lib \
+              --parallel \
+              --build_wheel \
+              --skip_submodule_sync \
+              --test --enable_onnx_tests --enable_transformers_tool_test \
+              --cmake_path /usr/bin/cmake --ctest_path /usr/bin/ctest"
+      workingDirectory: $(Build.SourcesDirectory)
+    displayName: 'Run onnxruntime unit tests'
+
+  - task: CmdLine@2
+    inputs:
+      script: |-
```

Comment:
Do we need to remove kernel_explorer test from rocm_training pipeline?

Reply:
Yes, if this new pipeline is put into use, we can then remove the dup ke test in training pipeline later.